ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PrivacyBlockProcessor.java

```diff
@@ -79,20 +80,23 @@ public Result processBlock(
       final MutableWorldState worldState,
       final BlockHeader blockHeader,
       final List<Transaction> transactions,
-      final List<BlockHeader> ommers) {
+      final List<BlockHeader> ommers,
+      final PrivateMetadataUpdater privateMetadataUpdater) {
+
+    if (privateMetadataUpdater != null) {
```

Comment:
I don't think I understand why we need this if statement. Is this because for each new block we need a new PrivateMetadataUpdater, and not reuse a previous one?

Reply:
I wasn't hundred percent sure either. But for every block that we are processing we do need a new PrivateMetadataUpdater. We are instantiating one in this block processor that is wrapping the public block processor.
If there was an updater passed in I thought that something must have gone wrong ...